modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/portlet/action/EditUserMVCActionCommand.java

```diff
@@ -180,6 +181,9 @@ protected void deleteUsers(ActionRequest actionRequest) throws Exception {
 
 					if (cmd.equals(Constants.DEACTIVATE)) {
 						status = WorkflowConstants.STATUS_INACTIVE;
+
+						_userGroupRoleLocalService.deleteUserGroupRolesByUserId(deleteUserId);
```

Comment:
Delete user roles is not a good way. Re-active deactivated user will be lost all roles.

Reply:
Hi @huynguyen-codeengine 
I updated my solution.
Just check the getAssigneesTotal() function. if a user is inactive will be not counted.
Please consider and let me know if it has any?